src/js/common/schemaform/widgets/CurrencyWidget.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+export default class CurrencyWidget extends React.Component {
+  onBlur = () => {
+    this.props.onBlur(this.props.id);
+  }
+
+  handleChange = (event) => {
+    const val = event.target.value;
+    if (val === '') {
+      this.props.onChange();
+    } else {
+      // Sometimes we get a string back (e.g. when you enter 10.00)
+      // so we need to parse it to be sure
+      const parsed = parseFloat(val);
+      if (!isNaN(parsed)) {
+        this.props.onChange(parsed);
+      } else {
+        this.props.onChange(val);
```

Comment:
I don't want to erase user input, if it all possible. I'd rather have a user enter "aaa" and see a validation error than have the input act like you're not typing in it.

This branch isn't actually hit much, since most (maybe all?) browsers return an empty string if the value isn't a number. Chrome won't even let you type non-numbers.